gooddata-java/src/main/java/com/gooddata/sdk/service/RequestIdInterceptor.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (C) 2004-2021, GoodData(R) Corporation. All rights reserved.
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE.txt file in the root directory of this source tree.
+ */
+package com.gooddata.sdk.service;
+
+import org.apache.commons.lang3.RandomStringUtils;
+import org.apache.http.Header;
+import org.apache.http.HttpException;
+import org.apache.http.HttpRequest;
+import org.apache.http.HttpRequestInterceptor;
+import org.apache.http.annotation.Contract;
+import org.apache.http.annotation.ThreadingBehavior;
+import org.apache.http.protocol.HttpContext;
+
+import java.io.IOException;
+
+import static com.gooddata.sdk.common.gdc.Header.GDC_REQUEST_ID;
+
+/**
+ * Intercepts the client-side requests on low-level in order to be able to catch requests also from the Sardine,
+ * that is working independently from Spring {@link org.springframework.web.client.RestTemplate} to set
+ * the X-GDC-REQUEST header to them.
+ */
+@Contract(threading = ThreadingBehavior.IMMUTABLE)
+public class RequestIdInterceptor implements HttpRequestInterceptor {
+
+    @Override
+    public void process(HttpRequest request, HttpContext context) throws HttpException, IOException {
+        StringBuilder requestIdBuilder = new StringBuilder();
+        Header requestIdHeader = request.getFirstHeader(GDC_REQUEST_ID);
+        if (requestIdHeader != null) {
+            requestIdBuilder.append(requestIdHeader.getValue()).append(":");
+        }
+        final String newRequestId = RandomStringUtils.randomAlphanumeric(16);
```

Comment:
I do not know what are requestid requirements, but shouldn't it be rathe some kind of UUID or hash? Also as far I remember the usual request ids are much shorter that 16 chars 🤔

Reply:
I took it from there https://github.com/gooddata/gdc-websupport/blob/79fee8f824566beca864b25221ab782cb5c43025/gdc-commons-web/src/main/java/com/gooddata/commons/web/filter/GdcCallContextFilter.java#L185.

Reply:
This is used also in Zendesk connectors.

Reply:
ok then